pkg/cstor/migrate/pool.go

```diff
@@ -0,0 +1,324 @@
+					cspiObj.Status.Phase)
+			}
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+	err = updateCVRsLabels(cspObj.Name, openebsNamespace, cspiObj)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// get csp for cspi on the basis of cspLabel, which is the combination of
+// hostname label on which cspi came up and the spc label.
+func getCSP(cspLabel string) (*apis.CStorPool, error) {
+	cspClient := csp.KubeClient()
+	cspList, err := cspClient.List(metav1.ListOptions{
+		LabelSelector: cspLabel,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(cspList.Items) != 1 {
+		return nil, fmt.Errorf("Invalid number of pools on one node: %d", len(cspList.Items))
+	}
+	cspObj := cspList.Items[0]
+	return &cspObj, nil
+}
+
+// The old pool pod should be scaled down before the new cspi pod comes up
+// to avoid importing the pool at two places at the same time.
+func scaleDownDeployment(cspObj *apis.CStorPool, openebsNamespace string) error {
+	klog.Infof("Scaling down deployemnt %s", cspObj.Name)
+	cspPod, err := pod.NewKubeClient().
+		WithNamespace(openebsNamespace).List(
+		metav1.ListOptions{
+			LabelSelector: "openebs.io/cstor-pool=" + cspObj.Name,
+		})
+	if err != nil {
+		return err
+	}
+	if len(cspPod.Items) > 0 {
+		_, err = deploy.NewKubeClient().WithNamespace(openebsNamespace).
+			Patch(
+				cspObj.Name,
+				types.StrategicMergePatchType,
+				[]byte(replicaPatch),
+			)
+		if err != nil {
+			return err
+		}
+		err = retry.
+			Times(60).
+			Wait(5 * time.Second).
+			Try(func(attempt uint) error {
+				_, err1 := pod.NewKubeClient().
+					WithNamespace(openebsNamespace).
+					Get(cspPod.Items[0].Name, metav1.GetOptions{})
+				if !k8serrors.IsNotFound(err1) {
+					return errors.Errorf("failed to get csp pod because %s", err1)
+				}
+				return nil
+			})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Update the bdc with the cspc labels instead of spc labels to allow
+// filtering of bds claimed by the migrated cspc.
+func updateBDCLabels(cspcName, openebsNamespace string) error {
+	bdcList, err := bdc.NewKubeClient().List(metav1.ListOptions{
+		LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + cspcName,
+	})
+	if err != nil {
+		return err
+	}
+	for _, bdcObj := range bdcList.Items {
+		klog.Infof("Updating bdc %s with cspc labels & finalizer.", bdcObj.Name)
+		delete(bdcObj.Labels, string(apis.StoragePoolClaimCPK))
+		bdcObj.Labels[string(apis.CStorPoolClusterCPK)] = cspcName
+		for i, finalizer := range bdcObj.Finalizers {
+			if finalizer == spcFinalizer {
+				bdcObj.Finalizers[i] = cspcFinalizer
+			}
+		}
+		// bdcObj.OwnerReferences[0].Kind = "CStorPoolCluster"
+		// bdcObj.OwnerReferences[0].UID = cspcObj.UID
+		_, err := bdc.NewKubeClient().
+			WithNamespace(openebsNamespace).
+			Update(&bdcObj)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update bdc %s with cspc label & finalizer", bdcObj.Name)
```

Comment:
I think we have to add a note to README.md. Once the user tried the migration process user should try until it succeeds or else there will be chances of stale BD's if the user triggers the deletion of the SPC process(Since CSPC is not yet created).

Reply:
Here, In the worst case, there will be a chance of network failures.